index.js

```diff
@@ -0,0 +1,66 @@
+const helmet = require('helmet');
+const morgan = require('morgan');
+const express = require('express');
+const bcrypt = require('bcryptjs');
+
+const db = require('./database/dbConfig.js');
+const server = express();
+
+server.use(express.json());
+server.use(morgan());
+server.use(helmet());
+
+
+server.get('/', (req, res) => {
+    res.send('My server can respond');
+});
+
+
+//==========CREATE A NEW USER(REGISTER)=========
+server.post('/api/register', (req, res) => {
+    //grabs username and password from body
+        const creds = req.body;
+
+    // generate hash from the user's password
+        const hash = bcrypt.hashSync(creds.password, 14);
+
+    //override the user.password with the hash
+        creds.password = hash;
+
+    //save the user to the database
+        db('users').insert(creds)
+        .then(ids => {
+        res.status(201).json(ids);
+        })
+        .catch(err => res.status(401).json(err));
```

Comment:
401 is not authorized, it's not clear if the failure was due to that reason. Could be that they didn't send in a password for example